Loop/Managers/WatchDataManager.swift

```diff
@@ -334,6 +372,52 @@ extension WatchDataManager {
 
         return items.joined(separator: "\n")
     }
+    
+    private func getSleepStartTime(start: Date, end: Date? = nil, sampleLimit: Int = 30, _ completion: @escaping (_ result: SleepStoreResult<Date>) -> Void) {
+        let predicate = HKQuery.predicateForSamples(withStart: start, end: end, options: [])
+        
+        getSleepStartTime(matching: predicate, sampleLimit: sampleLimit, completion)
+    }
+    
+    private func getSleepStartTime(matching predicate: NSPredicate, sampleLimit: Int, _ completion: @escaping (_ result: SleepStoreResult<Date>) -> Void) {
+        let sleepType = HKObjectType.categoryType(forIdentifier: HKCategoryTypeIdentifier.sleepAnalysis)!
+        // get more-recent values first
+        let sortDescriptor = NSSortDescriptor(key: HKSampleSortIdentifierEndDate, ascending: false)
+        
+        let query = HKSampleQuery(sampleType: sleepType, predicate: predicate, limit: sampleLimit, sortDescriptors: [sortDescriptor]) { (query, samples, error) in
+
+            if let error = error {
+                completion(.failure(error))
+            } else if let samples = samples as? [HKCategorySample] {
+                // find the average hour and minute components from the sleep start times
```

Comment:
Each sample's `value` property will correspond to a value of HKCategoryValueSleepAnalysis. If a user has a sleep tracking app, they might have primarily `.asleep` values; if the data comes from Bedtime in the Clock app, they'll have primarily `.inBed` values. It's possible that a user has a value of each type each night, in which case they'll both be averaged.

Consider performing an additional step here which filters samples down to those with `.inBed` values to use in the computation. If there aren't "enough", try `.asleep` instead. In any case, we'll want to make sure we don't include any `.awake` values.

Reply:
Done. For the purposes of "enough," what could be a decent threshold? The current code just needs one value for it to use the .inBed values instead of the .asleep values